### ISSUE TITLE
New Rule: table-column-count

### ISSUE THREAD

**Issue by SwetaTanwar** [eslint/markdown#389]
### Rule details

This rule is triggered when a GitHub Flavored Markdown table does not have the same number of cells in every row

### What type of rule is this?

Warns about a potential problem

### Example code

```markdown
This table's second data row has too few cells and its third data row has too many cells:

| Header | Header |
| ------ | ------ |
| Cell   | Cell   |
| Cell   |
| Cell   | Cell   | Cell   |
To fix these issues, ensure every row has the same number of cells:

| Header | Header |
| ------ | ------ |
| Cell   | Cell   |
| Cell   | Cell   |
| Cell   | Cell   |
```

### Participation

- [x] I am willing to submit a pull request to implement this rule.

### Additional comments

Prior Art: https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md056---table-column-count

**Comment by nzakas** [comment]
Seems like a good idea to me. :+1:
